wasmedge-ggml/llama-stream/src/main.rs

```diff
@@ -0,0 +1,185 @@
+use serde_json::Value;
+use std::collections::HashMap;
+use std::env;
+use std::io::{self, Write};
+use wasi_nn::{self, GraphExecutionContext};
+
+fn read_input() -> String {
+    loop {
+        let mut answer = String::new();
+        io::stdin()
+            .read_line(&mut answer)
+            .expect("Failed to read line");
+        if !answer.is_empty() && answer != "\n" && answer != "\r\n" {
+            return answer.trim().to_string();
+        }
+    }
+}
+
+fn set_data_to_context(
+    context: &mut GraphExecutionContext,
+    data: Vec<u8>,
+) -> Result<(), wasi_nn::Error> {
+    context.set_input(0, wasi_nn::TensorType::U8, &[1], &data)
+}
+
+#[allow(dead_code)]
+fn set_metadata_to_context(
+    context: &mut GraphExecutionContext,
+    data: Vec<u8>,
+) -> Result<(), wasi_nn::Error> {
+    context.set_input(1, wasi_nn::TensorType::U8, &[1], &data)
+}
+
+fn get_data_from_context(context: &GraphExecutionContext, index: usize, is_single: bool) -> String {
```

Comment:
It appears that `is_single` is always `true` in this example?

Reply:
Yes, but the commented code needs to call `get_metadata_from_context`. It will set the `is_single` to false.

Reply:
Oh. Okay. That is somewhat unexpected behavior. It means get_data and get_meta_data are not independent operations. But that that is okay.